### PR TITLE
fix(sdk): Update SDK to pull artifact related information from annotations

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Related #241 

**Description of your changes:**
As we move to multi-user for KFP, we want to inject certain artifact config via the annotations because we cannot change the Tekton API Spec. Thus, update the SDK to pull the related information from annotations without adding any breaking change.

Also we add few extra information to the annotations related to the artifact keys, so the UI can pull them based on similar requirements in #241. 

**Environment tested:**

* Python Version (use `python --version`): 3.7
* Tekton Version (use `tkn version`): 1.14.0
* Kubernetes Version (use `kubectl version`): 1.16
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

**Comment by drewbutlerbb4** [comment]
/lgtm
Thanks @Tomcli looks like it has everything needed from a UI perspective

File: sdk/python/kfp_tekton/compiler/_op_to_template.py [hunk 1]
```diff
@@ -464,8 +464,8 @@ def _op_to_template(op: BaseOp, pipelinerun_output_artifacts={}, enable_artifact
                 output_annotation = pipelinerun_output_artifacts.get(processed_op.name, [])
                 output_annotation.append(
                     {
-                        'name': output_artifact['name'],
-                        'path': output_artifact['path'],
+                        'name': output_artifact.get('name'),
```
Comment (ckadner) [inline sdk/python/kfp_tekton/compiler/_op_to_template.py:467]: can `output_artifact['name']` or `output_artifact['path']` ever be missing here (raising a `KeyError`)? If so, should the `get` return a default value like empty string `""`?

`output_artifact['name']` is used 2 lines down with direct key access as opposed to using `dict.get()`

Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/_op_to_template.py:467]: It was a merge from feng's PR yesterday, I can replace all of them to use .get() with defaults to make it consistent.

**Pull request merged**